### DISCUSSION TITLE
添加新的子集群，已经安装的 addon 或者 commponent 需要重新操作吗？

### DISCUSSION THREAD

**Discussion by skyshaokai** [kubevela/kubevela#4242] [Q&A]
添加新的子集群，已经安装的 addon 或者 commponent 需要重新操作吗？

**Comment by wonderflow** [comment]
addon 如果在子集群里也需要，需要重新 enable 一下

**Reply by miyoosan** [reply]
addon 如果disable再enable, 会比较费时，可以重启addon的工作流，比如对addon-fluxcd来说，执行`vela workflow restart addon-fluxcd -n vela-system`即可。它会自动把fluxcd addon的相关资源下发到新的子集群里。

**Reply by wonderflow** [reply]
Using vela workflow restart can cause unexpeced behavior, you should use this command in your scenario.

```
vela up addon-fluxcd -n vela-system --publish-version new-version
```

**Reply by wonderflow** [reply]
Ref https://github.com/kubevela/kubevela/pull/4839

**Reply by miyoosan** [reply]
好的，谢谢~~
我想进一步了解一下。
重新enable是指先disable再enable吗？如果disable，会影响正在running的application么？

比如：我使用addon-fluxcd来实现gitops功能，如果disable了addon-fluxcd，那gitops功能受影响吗？我不太希望因为新加了子集群，为了让子集群拥有addon-fluxcd插件的能力，就必须去中断已有集群的gitops功能。

`vela up addon-fluxcd -n vela-system --publish-version new-version` 这个命令支持我这类诉求么？

**Reply by wonderflow** [reply]
你的场景不要disable再enable，会影响运行的应用。 Vela up那个命令满足你说的场景，不会影响已经启动的集群

**Reply by miyoosan** [reply]
> 你的场景不要disable再enable，会影响运行的应用。 Vela up那个命令满足你说的场景，不会影响已经启动的集群

好的，谢谢！请问是在多少版本以后支持这个命令呢？new-version这个版本号是要跟已安装的插件版本号保持一致的吗?

**Reply by wonderflow** [reply]
new-version 这个版本号你可以自己指定，这个命令在 1.4+ 中已经有了。

**Reply by miyoosan** [reply]
OKOK，没有疑问了，谢谢你！
